code-for-canada/project-thundercat
# [Test API] Test Access - Public and Non-Public Tests

# Description

This PR is introducing basic conditions that manages the test accesses. Default users can only access public tests and admin users can access public and non-public tests.

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshot

**Public Test (Authenticated or not):**

![image](https://user-images.githubusercontent.com/23021242/59266547-df4bcf00-8c15-11e9-88d5-a135d9b73729.png)

**Non-Public Test (Not authenticated):**

![image](https://user-images.githubusercontent.com/23021242/59266606-07d3c900-8c16-11e9-8751-12581d669894.png)

**Non-Public Test (Authenticated as default user):**

![image](https://user-images.githubusercontent.com/23021242/59266644-1de18980-8c16-11e9-905e-2c8eac46d03c.png)

**Non-Public Test (Authenticated as admin user):**

![image](https://user-images.githubusercontent.com/23021242/59266677-30f45980-8c16-11e9-9097-49014a130832.png)

# Testing

Manual steps to reproduce this functionality:

1.  Try to access public and non-public tests.
2.  Make sure that you have the right accesses depending on the account you are using (default or admin).

# Checklist
